ext/spl/spl_directory.c

```diff
@@ -2576,11 +2577,15 @@ SPL_METHOD(SplFileObject, fgetcsv)
 		switch(ZEND_NUM_ARGS())
 		{
 		case 3:
-			if (esc_len != 1) {
-				php_error_docref(NULL, E_WARNING, "escape must be a character");
+			if (esc_len > 1) {
+				php_error_docref(NULL, E_WARNING, "escape must be empty or a single character");
 				RETURN_FALSE;
 			}
-			escape = esc[0];
+			if (esc_len == 0) {
+				escape = EOF;
```

Comment:
The use of `EOF` for this purpose feels rather arbitrary. I'd prefer to define an explicit value for this, or to have a separate argument for whether it is enabled.

An alternative would be to use the value `'\0'` for this -- I believe this is already a de-facto standard for disabling escaping as it is a character that does not typically occur inside CSV files.

Reply:
I've written a [respective RFC](https://wiki.php.net/rfc/kill-csv-escaping).